src/helpers/daily-data-types/combined.tsx

```diff
@@ -92,12 +84,15 @@ let combinedTypeDefinitions: DailyDataTypeDefinition[] = [
         type: DailyDataType.MindfulMinutes,
         dataProvider: combinedMindfulMinutesDataProvider,
         availabilityCheck: async (modifiedAfter?: Date): Promise<boolean> => {
-            const result = await simpleAvailabilityCheck("AppleHealth", "MindfulSession")(modifiedAfter);
-            if (!result) {
-                return simpleAvailabilityCheck("GoogleFit", "ActivitySegment")(modifiedAfter);
-            } else {
-                return result;
+            const settings = await MyDataHelps.getDataCollectionSettings();
+
+            if (settings.queryableDeviceDataTypes.find(dt => dt.namespace == "AppleHealth" && dt.type == "MindfulSession")
+                && await simpleAvailabilityCheck("AppleHealth", "MindfulSession")(modifiedAfter)) {
+                return true;
             }
+
+            return !!settings.queryableDeviceDataTypes.find(dt => dt.namespace == "GoogleFit" && dt.type == "ActivitySegment")
```

Comment:
Here and on line 114 you did an explicit boolean coercion, but nowhere else that you're doing the same type of `find` call.  Is there something different about these two spots?

Reply:
Yes.  This confused me at first, too (when the IDE suggested it).  It has to do with how the value from the `&&` evaluation works with ifs vs returns.

The resulting value from `obj && bool` is:
- `obj` when `obj` is falsy  (this is the important part)
- `bool` when `obj` is truthy

Since `obj` will only be falsy when it is `undefined`, the returned data type from the `&&` condition is `undefined | boolean`.  This works fine within the `if` conditions, because that `undefined | boolean` value gets re-evaluated for truthy-ness.  However, when it is being used as a return value that needs to be typed as a `boolean`, it doesn't match.  The `!!` operator forces it to be a `bool && bool` condition, which has the correct data type for the return statements.

Reply:
Ah! Appreciate the explanation.  Makes sense - I didn't think to delete the `!!` locally in my IDE and see what happened because I figured it was a style choice and not a compilation one!